src/mlpack/methods/ann/ffn.hpp

```diff
@@ -159,8 +150,8 @@ class FFN
    * @return The final objective of the trained model (NaN or Inf on error).
    */
   template<typename OptimizerType = ens::RMSProp, typename... CallbackTypes>
-  double Train(arma::mat predictors,
-               arma::mat responses,
+  double Train(InputType predictors,
+               OutputType responses,
```

Comment:
Shouldn't this also be InputType responses?

Reply:
Yes, nice catch!